src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityUtils.cs

```diff
@@ -216,6 +216,25 @@ internal static ReadOnlyCollection<SecurityKey> CreateSymmetricSecurityKeys(byte
             return temp.AsReadOnly();
         }
 
+        internal static SecurityKeyIdentifier CreateSecurityKeyIdentifier(Microsoft.IdentityModel.Xml.KeyInfo keyInfo)
+        {
+            if(keyInfo!=null && keyInfo.RSAKeyValue !=null)
+            {
+                throw new NotSupportedException("RSA key not supported.");
+            }
+
+            foreach (var objdata in keyInfo.X509Data)
+            {
+                foreach (String certificateStr in objdata.Certificates)
+                {
+                    byte[] data = Convert.FromBase64String(certificateStr);
+                    return new SecurityKeyIdentifier(new X509RawDataKeyIdentifierClause(data));;
+                }
```

Comment:
There's a constructor which takes a bool cloneBuffer parameter. Use that overload and pass false as the default is true which allocates a new buffer and copies the data across. The purpose of this is when the buffer has come from a buffer pool and might be modified. In this case, X509RawDataKeyIdentifierClause can take ownership of this allocated buffer.